File: Cplusplus/_0000_study/_common/Output.cpp
#include <iostream>
#include <vector>

using namespace std;

void OutputBasicArray1D(vector<int> nums) {
    cout << "[";
    for (int i = 0; i < nums.size(); i++) {
        cout << nums[i];
        if (i != nums.size() - 1) cout << ", ";
    }
    cout << "]" << endl;
}

void OutputBasicArray2D1(vector<vector<int>> nums) {

    cout << "[";
    for (int i = 0; i < nums.size(); i++) {
        cout << "[";
        for (int j = 0; j < nums[i].size(); j++) {
            cout << nums[i][j];
            if (j != nums[i].size() - 1) cout << ", ";
        }
        cout << "]";
        if (i != nums.size() - 1) cout << ", ";
    }
    cout << "]" << endl;
}

void OutputBasicArray2D(vector<vector<int>> nums) {

    cout << "[";
    for (int i = 0; i < nums.size(); i++) {
        if (i > 0 && i < nums.size()) cout << ' ';
        cout << "[";
        for (int j = 0; j < nums[i].size(); j++) {
            cout << nums[i][j];
            if (j != nums[i].size() - 1) cout << ", ";
        }
        cout << "]";
        if (i != nums.size() - 1) cout << ", " << endl;
    }
    cout << "]" << endl;
}

int main() {

    vector<vector<int>> nums = {{1, 2},
                                {2, 4, 6}};
    OutputBasicArray2D1(nums);
    OutputBasicArray2D(nums);

    return 0;
}

File: Java/src/_0000_study/_common/Output.java
package _0000_study._common;

import java.util.Arrays;
import java.util.stream.Collectors;

/**
 * Date: 2020/11/10 9:51
 * Content: 用于输出各种格式的数据
 */
public class Output {

    public static void OutputBasicArray1D(int[] nums) {

        System.out.println(Arrays.toString(nums));
    }


    public static void OutputBasicArray2D(int[][] nums) {

        System.out.print("[");
        for (int i = 0; i < nums.length; i++) {
            if (i == 0)
                System.out.println(Arrays.toString(nums[i]) + ",");
            else if (i < nums.length - 1)
                System.out.println(" " + Arrays.toString(nums[i]) + ",");
            else
                System.out.print(" " + Arrays.toString(nums[i]));
        }
        System.out.print("]");
    }

    public static void OutputBasicArray2D(char[][] nums) {

        System.out.print("[");
        for (int i = 0; i < nums.length; i++) {
            if (i == 0)
                System.out.println(Arrays.toString(nums[i]) + ",");
            else if (i < nums.length - 1)
                System.out.println(" " + Arrays.toString(nums[i]) + ",");
            else
                System.out.print(" " + Arrays.toString(nums[i]));
        }
        System.out.print("]");
    }

    public static void OutputBasicArray2D1(int[][] nums) {

        System.out.print("[");
        for (int i = 0; i < nums.length; i++) {
            if (i != nums.length - 1)
                System.out.print(Arrays.toString(nums[i]) + ", ");
            else
                System.out.print(Arrays.toString(nums[i]));
        }
        System.out.print("]");
    }
}
